src/main/java/com/vladbudan/restapp/servlet/MainServlet.java

```
@@ -0,0 +1,516 @@
+            if (action == null) {
+                createUser(request, response);
+            } else if (action.endsWith("/cats")) {
+                createCat(request, response);
+            } else if (action.endsWith("/dogs")) {
+                createDog(request, response);
+            }
+        } catch (IOException | ParseException e) {
+            throw new MyException(e.getMessage());
+        }
+
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest request, HttpServletResponse response) throws MyException {
+
+        try {
+            String action = request.getPathInfo();
+            if (action == null) {
+                updateUser(request, response);
+            } else if (action.endsWith("/cats")) {
+                updateCat(request, response);
+            } else if (action.endsWith("/dogs")) {
+                updateDog(request, response);
+            }
+        } catch (IOException | ParseException e) {
+            throw new MyException(e.getMessage());
+        }
+
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws MyException {
+
+        try {
+            String action = request.getPathInfo();
+            if (action == null) {
+                deleteUser(request, response);
+            } else if (action.endsWith("/cats")) {
+                deleteCat(request, response);
+            } else if (action.endsWith("/dogs")) {
+                deleteDog(request, response);
+            }
+
+        } catch (IOException e) {
+            throw new MyException(e.getMessage());
+        }
+
+    }
+
+    private void getUser(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        int userId = Integer.parseInt(request.getParameter("id"));
+        User user = userService.getById(userId);
+
+        if (user == null) {
+            throw new ResourceNotFoundException("User with such id not found!");
```

You should throw this exception in the service, not in the controller